Create List, Destroy, Provider and createReducer documentation (#39)
SeasonedSoftware/croods: src/docs/Destroy.mdx

---
name: Destroy
route: /destroy
---

import { Playground } from 'docz'

# Destroy

> The Destroy component provides a easy way to destroy an entry in the server, using the response as needed.

Reviewer: ```suggestion
The Destroy component provides an easy way to destroy an entry in the server, and gives the server's response for convenience.
```

## API

| Property | Type          | Required | Default    | Description                                                                                             |
| -------- | ------------- | -------- | ---------- | ------------------------------------------------------------------------------------------------------- |

> | name     | String        | true     | -          | It defines which reducer will be used and, if no path have been provider, will set path to baseUrl/name |

Reviewer: ```suggestion
| name     | String        | true     | -          | It defines which reducer will be used and will be used as `path` if none is provided:  ex: `baseUrl/name` |
```

> | id       | String/Number | true     | -          | It defines the id of the entry to be destroyed, passing it in the url defined in path                   |

Reviewer: ```suggestion
| id       | String/Number | true     | -          | It defines the ID of the entry to be destroyed, passing it in the url defined in path                   |
```

| render   | Func          | true     | -          | A function returning a React Node. Ex: ({ list }, props) -> < JSX />                                    |

> | parentId | String/Number | false    | -          | Parent element's Id, for more specificity in changes.                                                   |

Reviewer: ```suggestion
| parentId | String/Number | false    | -          | Parent element's ID, for nested resources.                                                   |
```

> | path     | String        | false    | `/${name}` | Ex: /foo/bar-\_12?&=                                                                                    |

Reviewer: ```suggestion
| path     | String        | false    | `/${name}` | Ex: /foo/bar?&=                                                                                    |
```

## How to use

```javascript
import { Destroy } from 'croods'

export default (
  <Destroy
    name="pets"
    id="123"
    // if path is different than /pets/${id}, you can set a custom path like so:
    // path='/customPath'
    render={destroy => <DestroyButton destroy={destroy} />}
  />
)
```

### render

> The render function is used to render a component passing to it a **destroy** function, and

Reviewer: ```suggestion
The render function is used to render a component (ex: a trash bin icon) passing to it a **destroy** function you can use to dispatch the request (ex: when trash icon is clicked).
```
